spec/lib/data_explorer_query_group_bookmarkable_spec.rb

```diff
@@ -0,0 +1,153 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+describe DataExplorerQueryGroupBookmarkable do
+  fab!(:admin_user) { Fabricate(:user) }
+  fab!(:user) { Fabricate(:user) }
+  fab!(:guardian) { Guardian.new(user) }
+  fab!(:group0) { Fabricate(:group)}
+  fab!(:group1) { Fabricate(:group)}
+  fab!(:group2) { Fabricate(:group)}
+  fab!(:group3) { Fabricate(:group)}
+  fab!(:query1) { Fabricate(:query, name: "My First Query", 
+                                    description: "This is the description of my 1st query.",
+                                    sql: "Not really important",
+                                    user: admin_user) }
+  fab!(:query2) { Fabricate(:query, name: "My Second Query", 
+                                    description: "This is my 2nd query's description.",
+                                    sql: "Not really important",
+                                    user: admin_user) }
+  
+  before do
+    Bookmark.register_bookmarkable(DataExplorerQueryGroupBookmarkable)
+  end
```

Comment:
Just curious, do we need to register this again since we've already registered it in `plugin.rb`?

Reply:
Because the registered bookmarkables are reset between specs. I know it's kind of awkward, not sure how to get around it though:

https://github.com/discourse/discourse/blob/3941bad0751fc2e59576d77bef531b0088571104/spec/rails_helper.rb#L143-L145

Reply:
@tgxworld

Reply:
I wonder if we will be able to snapshot the bookmarkable registered at the start of running the suite and then roll back each tests to those bookmarkable at the end of each test.